e2e/config/azure.go

```diff
@@ -126,37 +134,74 @@ func NewAzureClient(subscription string) (*AzureClient, error) {
 
 	cloud.VMSS, err = armcompute.NewVirtualMachineScaleSetsClient(subscription, credential, opts)
 	if err != nil {
-		return nil, fmt.Errorf("failed to create vmss client: %w", err)
```

Comment:
nit: fail/failed are keywords that are normally used when grepping for what went wrong. I"m not sure how practical it is to remove them here, I would personally keep them

Reply:
I avoid including `error` or `failed` in the error message. It just produces a lot of duplication and noise when you wrap errors multiple times:

```
ERROR: error create client: error create VMSS client: error create something: error do something
```
vs

```
ERROR: create client: create VMSS client: create something: do something
```

I don't think it's hard to search them, but it improves the message itself when you surface it (sometimes accidental)

Reply:
It also helps if you ever want to apply a fancy formatting to an error. I think it's a common recommendation to omit the keywords.

Reply:
makes sense if the logger prints the level, this is indeed redundant.